Put network protocol version in the demo header

sprunk opened the pull request:
See #2140.

Something to figure out would be how to tell autohosts this, since either modifying `SERVER_STARTED` or sending a different packet would be a breaking change. Perhaps just send two packets (`SERVER_STARTED` as-is, plus a new future-compatible one with metadata)?

p2004a reviewed (approved):
This change looks good. For autohost I would change the `SERVER_STARTED` adding a field to it. It is one breaking change and IMHO that's the point: we switch from "magic frozen autohost interface" to "able to make changes and communicate".

sprunk committed:
Move `NETWORK_VERSION` to `SpringVersion` namespace

Will let it be more widely used in various other places that want it.

sprunk committed:
`NETWORK_VERSION`: engine major -> 2027

This value is supposed to represent the version of the network/replay
packet protocol.

It used to be hardcoded to the major version, which historically made
sense when the engine used approximately semantic versioning and most
games only used stable releases.

Nowadays the engine uses chronological versioning and games regularly
use dev versions, so the above is insufficient.

Note that while this will eventually help various tools (replay parsers,
autohosts, dedi etc) it will require some maintenance vigilance to make
sure it actually gets incremented (which until now happened "passively").

Start the new scheme at 2027, which is a bit beyond the current major.

sprunk committed:
Put the network protocol version in the demo header.

sprunk committed:
Autohost interface: add network version to SERVER_STARTED

sprunk commented on rts/System/LoadSave/demofile.h:
```diff
@@ -89,6 +90,7 @@ struct DemoFileHeader
 		swabDWordInPlace(teamStatElemSize);
 		swabDWordInPlace(teamStatPeriod);
 		swabDWordInPlace(winningAllyTeamsSize);
+		swabDWordInPlace(networkVersion);
```
I see https://github.com/beyond-all-reason/demo-parser/issues/38 is fixed, is this ready to go now?

p2004a replied:
For bar, the parser updat also needs to be released in spring-launcher. There are no releases of it for a long time. I guess convince TK to prioritize it? (I'm personally not looking forward to do this thankless irksome task)

sprunk replied:
cc @lostsquirrel1 ^ (but no hurry)
